erpnext/controllers/subcontracting.py

```diff
@@ -0,0 +1,342 @@
+from __future__ import unicode_literals
+
+import frappe
+from frappe import _
+from frappe.utils import flt, cint
+from collections import defaultdict
+from erpnext.stock.doctype.serial_no.serial_no import get_serial_nos
+
+class Subcontracting(object):
+	def set_materials_for_subcontracted_items(self, raw_material_table):
+		if self.doctype == 'Purchase Invoice' and not self.update_stock:
+			return
+
+		self.raw_material_table = raw_material_table
+		self.identify_change_in_item_table()
+		self.prepare_supplied_items()
+		self.validate_consumed_qty()
+
+	def prepare_supplied_items(self):
+		self.initialized_fields()
+		self.get_purchase_orders()
+		self.get_pending_qty_to_receive()
+		self.get_available_materials()
+		self.remove_changed_rows()
+		self.set_supplied_items()
+
+	def initialized_fields(self):
+		self.available_materials = frappe._dict()
+		self.alternative_item_details = frappe._dict()
+		self.get_backflush_based_on()
+
+	def get_backflush_based_on(self):
+		self.backflush_based_on = frappe.db.get_single_value("Buying Settings",
+			"backflush_raw_materials_of_subcontract_based_on")
```

Comment:
Could add this in class `init` so it doesn't have to be invoked
```py
def __init__(self):
    self.backflush_based_on = frappe.db.get_single_value("Buying Settings", 
        "backflush_raw_materials_of_subcontract_based_on")

def set_materials_for_subcontracted_items(self, raw_material_table):
    ...
```

Reply:
__init__ will call every time even though the order is not subcontracted